sound/smooth_swing_v2.h

```diff
@@ -7,220 +7,243 @@
+                        STDOUT.print(A.end());
+                        STDOUT.print("  mixhum: ");
+                        STDOUT.print(mixhum);
+                        STDOUT.print("  mixab: ");
+                        STDOUT.print(mixab);
+                        STDOUT.print("  hum_volume: ");
+                        STDOUT.println(hum_volume);
+                    }
+                    if (on_) {
+                        // We need to stop setting the volume when off, or playback may never stop.
+                        A.set_volume(mixhum * mixab);
+                        B.set_volume(mixhum * (1.0 - mixab));
+                    }
+                    break;
+                }
+                A.set_volume(0);
+                B.set_volume(0);
+                state_ = SwingState::OUT;
+                
+            case SwingState::OUT:
+                if (!A.isOff() || !B.isOff()) {
+                    if (monitor.ShouldPrint(Monitoring::MonitorSwings)) {
+                        Serial.println("Waiting for volume = 0");
+                    }
+                }
+                PickRandomSwing();
+                state_ = SwingState::OFF;
+        }
+        // Must always set hum volume, or fade-out doesn't work.
+        delegate_->SetHumVolume(hum_volume);
     }
-    RefPtr<BufferedWavPlayer> player;
-    float midpoint = 0.0;
-    float width = 0.0;
-  };
-  Data A;
-  Data B;
-
-  uint32_t last_random_ = 0;
-  bool on_ = false;;
-  BoxFilter<Vec3, 3> gyro_filter_;
-  int swings_;
-  uint32_t last_micros_;
-  SwingState state_ = SwingState::OFF;;
+    
+private:
+    struct Data {
+        void set_volume(float v) {
+            if (player) player->set_volume(v);
+        }
+        void Play(Effect* effect, float start = 0.0) {
+            if (!player) {
+                player = GetFreeWavPlayer();
+                if (!player) return;
+            }
+            player->set_volume(0.0);
+            player->PlayOnce(effect, start);
+            player->PlayLoop(effect);
+        }
+        void Off() {
+            if (!player) return;
+            player->set_fade_time(0.2);  // Read from config file?
+            player->FadeAndStop();
+        }
+        void Free() {
+            if (!player) return;
+            player.Free();
+        }
+        void Stop() {
+            if (!player) return;
+            player->Stop();
+        }
+        bool isOff() {
+            if (!player) return true;
+            return player->isOff();
+        }
+        void SetTransition(float mp, float w) {
+            midpoint = mp;
+            width = w;
+        }
+        float begin() const { return midpoint - width / 2; }
+        float end() const { return midpoint + width / 2; }
+        void rotate(float degrees) {
+            midpoint += degrees;
+        }
+        RefPtr<BufferedWavPlayer> player;
+        float midpoint = 0.0;
+        float width = 0.0;
+    };
+    Data A;
+    Data B;
+    BufferedWavPlayer D;
```

Comment:
We don't want a whole new bufferedwavplayer here.
Besides, it won't do anything unless hooked up to the dynamic mixer.

We just need to do what HybridFont::PlayPolyphonic does.
Alternatively, we could just allow swing events go down to the underlying font, and let it make up it's own mind of when to play swing sounds.